src/olympia/addons/tests/test_utils_.py

```diff
@@ -278,3 +278,66 @@ def test_lwt_missing_info(self):
                 u'#000')
             assert (xpi.read('weta.png') ==
                     open(self.background_png).read())
+
+
+class TestBuildWebextDictionaryFromLegacy(AMOPaths, TestCase):
+    def setUp(self):
+        self.addon = addon_factory(
+            target_locale='ar', type=amo.ADDON_DICT,
+            version_kw={'version': '1.0'},
+            file_kw={'is_webextension': False})
+        self.xpi_copy_over(
+            self.addon.current_version.all_files[0], 'dictionary-test.xpi')
+
+    def check_xpi_file_contents(self, xpi_file_path, expected_version):
+        addon = self.addon
+        with zipfile.ZipFile(xpi_file_path, 'r', zipfile.ZIP_DEFLATED) as xpi:
+            # Check that manifest is present, contains proper version and
+            # dictionaries properties.
+            manifest = xpi.read('manifest.json')
+            manifest_json = json.loads(manifest)
+            assert manifest_json['version'] == expected_version
+            expected_dict_obj = {addon.target_locale: 'dictionaries/ar.dic'}
+            assert manifest_json['dictionaries'] == expected_dict_obj
+
+            # Check that we haven't included any useless files.
```

Comment:
where do we add useless files?

Reply:
The file we're testing with (`src/olympia/files/fixtures/files/dictionary-test.xpi`) already has an `install.js` file as well as a `__MACOSX` directory. And obviously, an `install.rdf` file.